infra/module-swapper/cmd/swap.go

```diff
@@ -54,28 +66,30 @@ func getModuleNameRegistry(dir string) (string, string, error) {
 	if err != nil {
 		return "", "", err
 	}
-
-	// GH remote will be of form https://github.com/ModuleRegistry/ModuleName
-	if !strings.Contains(remote, "https://github.com/") {
-		return "", "", fmt.Errorf("Expected GitHub remote of form https://github.com/ModuleRegistry/ModuleRepo. Got: %s", remote)
+	u, err := giturl.Parse(remote)
+	if err != nil {
+		return "", "", err
+	}
+	if u.Host != "github.com" {
+		return "", "", fmt.Errorf("expected GitHub remote, got: %s", remote)
 	}
+	orgRepo := u.Path
+	orgRepo = trimAnySuffixes(orgRepo, []string{"/", ".git"})
+	orgRepo = strings.TrimPrefix(orgRepo, "/")
 
-	// remotes maybe suffixed with a trailing / or .git
-	remote = trimAnySuffixes(remote, []string{"/", ".git"})
-	namePrefix := strings.ReplaceAll(remote, "https://github.com/", "")
-	if !strings.Contains(namePrefix, "/") {
-		return "", "", fmt.Errorf("Expected GitHub org/owner of form ModuleRegistry/ModuleRepo. Got: %s", namePrefix)
+	split := strings.Split(orgRepo, "/")
+	if len(split) != 2 {
+		return "", "", fmt.Errorf("expected GitHub remote of form https://github.com/ModuleRegistry/ModuleRepo, got: %s", remote)
 	}
-	moduleRegistry := namePrefix[:strings.LastIndex(namePrefix, "/")]
-	repoName := namePrefix[strings.LastIndex(namePrefix, "/")+1:]
+	org, repoName := split[0], split[1]
 
 	// module repos are prefixed with terraform-google-
```

Comment:
Do we want the option to use with non-confirming repo names?  https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/terraform/test-org/org/locals.tf#L120

Reply:
Just saw these:
- https://github.com/terraform-google-modules/terraform-example-foundation/pull/1064
- https://github.com/terraform-google-modules/terraform-example-foundation/commit/bed56d961ba45dcf36a60d2ebc5e55fad5b3389e#commitcomment-135386931

Reply:
Sounds like at a minimum we might want to instead exit gracefully and log that the repo is not supported, etc.

Reply:
Yup! Fixed this